src/mlmc/new_simulation.py

```diff
@@ -0,0 +1,80 @@
+import numpy as np
+import attr
+from abc import ABC, abstractmethod
+from typing import List, Dict, Tuple, Optional, Any
+
+
+@attr.s(auto_attribs=True)
+class QuantitySpec:
+    name: str
+    unit: str
+    shape: Tuple[int, int]
+    times: List[float]
+    locations: List[str]
```

Comment:
Would be great if we can have either List[str] or List[Tuple[float, float, float]] here, i.e.
Union of these two.